### PR TITLE
NMS-13754: Daemon config file doc update

### PR DESCRIPTION
**Pull request by mmahacek**
### All Contributors

* [X] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [X] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://issues.opennms.org) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the JIRA issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here,
or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-13754

### CODE REVIEW

**Review by Bonrob2 — COMMENTED** [review]
A few changes. I notice some daemons are missing descriptions. Can we create a ticket to add these later?

File: docs/modules/reference/pages/daemons/daemon-config-files/collectd.adoc [hunk 1]
```diff
@@ -3,31 +3,103 @@
...
+| no
+| yes
+
+| prometheus-datacollection.d/*.xml
+| Defines data collections via Prometheus Exporter.
+| no
+| yes
+
+| snmp-config.xml
+| Defines data per-host security settings for SNMP communication.
```
[the remark this answers is not in the document]
Comment (mmahacek) [reply docs/modules/reference/pages/daemons/daemon-config-files/collectd.adoc:66]:
```suggestion
| Defines per-host security settings for SNMP communication.
```

File: docs/modules/reference/pages/daemons/daemon-config-files/collectd.adoc [hunk 2]
```diff
@@ -3,31 +3,103 @@
...
+| no
+| no
+
+| wmi-config.xml
+| Defines data per-host security settings for WMI communication.
```
[the remark this answers is not in the document]
Comment (mmahacek) [reply docs/modules/reference/pages/daemons/daemon-config-files/collectd.adoc:71]:
```suggestion
| Defines per-host security settings for WMI communication.
```

File: docs/modules/reference/pages/daemons/daemon-config-files/collectd.adoc [hunk 3]
```diff
@@ -3,31 +3,103 @@
...
+| no
+| no
+
+| wmi-datacollection-config.xml
+| Defines data collections via WMI.
+| no
+| yes
+
+| wsman-config.xml
+| Defines data per-host security settings for WSMAM communication.
```
[the remark this answers is not in the document]
Comment (mmahacek) [reply docs/modules/reference/pages/daemons/daemon-config-files/collectd.adoc:81]:
```suggestion
| Defines per-host security settings for WSMAM communication.
```

File: docs/modules/reference/pages/daemons/introduction.adoc [hunk 4]
```diff
@@ -4,38 +4,99 @@
...
+| Enhanced link daemon (Enlinkd) discovers connections between nodes using data generated by various link discovery protocols and accessible via SNMP.
+
+| EventTranslator
+| The Event Translator (ET) can transform and enhance the data from OpenNMS events that other daemons generate.
+
+| xref:daemons/daemon-config-files/eventd.adoc[Eventd]
+|
+
+| JettyServer
+| The Jetty server provides the web user interface of OpenNMS.
```
[the remark this answers is not in the document]
Comment (mmahacek) [reply docs/modules/reference/pages/daemons/introduction.adoc:44]:
```suggestion
| The Jetty server provides the {page-component-title} web user interface.
```

**Review by Bonrob2 — APPROVED** [review]
LGTM (see below. I did notice a small typo)

File: docs/modules/reference/pages/daemons/daemon-config-files/telemetryd.adoc [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+| yes
+| no
+
+| org.opennms.features.flows.classification.cfg
+| Rules for classifying flow data.
+| yes
+| no
+
+| org.opennms.features.flows.persistence.elastic.cfg
+| Configuration for offloading flow data to Elasticsearch/
```
Comment (Bonrob2) [inline docs/modules/reference/pages/daemons/daemon-config-files/telemetryd.adoc:30]:
```suggestion
| Configuration for offloading flow data to Elasticsearch.
```

**Review by Bonrob2 — COMMENTED** [review]
Noticed a small typo.

**Review by Bonrob2 — DISMISSED** [review]
LGTM!

**Review by dino2gnt — APPROVED** [review]
Looks fine.  No specific feedback.

**Review by RangerRick — APPROVED** [review]

**Pull request merged**